content/en/blog/_posts/2024-06-20-pod-failure-policy-ga

```diff
@@ -0,0 +1,124 @@
+   (`.backoffLimit` or `.backoffLimitPerIndex` if specified)
+
+### Pod conditions
+
+A significant part of the development work was for this feature was to introduce
+the built-in `DisruptionTarget` condition to annotate reasons for Pod failures
+due to disruptions. The condition can have the following values of the `reason`
+field, depending on the scenario:
+
+- `PreemptionByKubeScheduler` - Pod preempted by kube-scheduler
+- `DeletionByTaintManager` - Pod evicted by kube-controller-manager due to taints
+- `EvictionByEvictionAPI` - Pod deleted by Eviction API
+- `DeletionByPodGC` - an orphaned Pod deleted by pod GC
+- `TerminationByKubelet` - Pod terminated due to graceful node shutdown or preemption for critical pods.
+
+Notably, Pod failure policy for Jobs can be used against custom user-supplied
+conditions, either added by webhooks or external controllers.
+
+Users may use them to annotate for:
+- annotating user-specific Pod failures, such as configuration issues,
+- annotating Kubernetes failure scenarios which are currently not yet recognized
+  by any built-in conditions (e.g. admission failures due to `OutOfPods`).
+
+### Example
+
+The Pod failure policy snippet below demonstrates how to use to use it to:
+- ignore any Pod disruption recognized by the built-in DisruptionTarget condition
+- fail the Job in case of configuration issues annotated by the custom `ConfigIssue`
+- fail the Job in case of Pod failure with the reserved exit code 42
+- count all other failures towards the default `backoffLimit` (or
+  `backoffLimitPerIndex` if used)
+
+```yaml
+podFailurePolicy:
+  rules:
+  - action: Ignore
+    onPodConditions:
+    - type: DisruptionTarget
+  - action: FailJob
+    onPodConditions:
+    - type: ConfigIssue
+  - action: FailJob
+    onExitCodes:
+      operator: In
+      values: [ 42 ]
+```
+
+## How can you learn more?
+
+- Read the user-facing documentation for
+  [Pod failure policy](/docs/concepts/workloads/controllers/job/#pod-failure-policy) and
+  [Backoff limit per index](/docs/concepts/workloads/controllers/job/#backoff-limit-per-index)
+- Read the tasks page on
+  [Handling retriable and non-retriable pod failures with Pod failure policy](https://kubernetes.io/docs/tasks/job/pod-failure-policy/)
+- Read the KEP for [Pod failure policy](https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3329-retriable-and-non-retriable-failures)
+
+## How to get involved?
+
+This work was sponsored by
+[batch working group](https://github.com/kubernetes/community/tree/master/wg-batch)
+in close collaboration with the
+[SIG Apps](https://github.com/kubernetes/community/tree/master/sig-apps),
+and [SIG Node](https://github.com/kubernetes/community/tree/master/sig-node),
+and [SIG Scheduling](https://github.com/kubernetes/community/tree/master/sig-scheduling)
+communities.
+
+If you are interested in working on new features in the space we recommend
```

Comment:
Maybe we should mention sig-apps slack also?

And should you have some acknowledgements?

Reply:
> And should you have some acknowledgements?

Thank you for bringing it up. I have tons of people I would love to thank to for reviews and feedback! The only challenge is how not to forget anyone :) I will try my best to come up with something!

Reply:
Added the section. Again, hope I didn't missed anyone, this is my best effort attempt. If there is someone who feels missed please ping me on slack, it is very possible I forgot someone's contribution